packages/storybook/.storybook/main.js

```diff
@@ -1,5 +1,27 @@
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const { VanillaExtractPlugin } = require("@vanilla-extract/webpack-plugin");
+const path = require("path");
+
 module.exports = {
   stories: ["../stories/**/*.stories.*"],
   addons: ["@storybook/addon-links", "@storybook/addon-essentials", "storybook-addon-themes"],
   framework: "@storybook/react",
+  core: {
+    builder: "webpack5",
+  },
+  webpackFinal: (baseConfig) => {
+    const { module = {}, plugins = {} } = baseConfig;
+
+    // by default, the rules for jsx|tsx files are including the whole monorepo root,
+    // but it's enough to check only the storybook package
+    const tsxRules = module.rules.filter((rule) => rule?.test?.test("test.tsx"));
```

Comment:
Not sure if this was the cause for the hang during the storybook-start, but there's no need to include the whole monorepo root. We just want to transpile the files in the storybook package

Reply:
re-tested without these lines and the issue reappears. I suppose storybook is trying to react both to the change in the library itself, and at the same time to the change in the node_modules after the library is built by tsup